lib/srv/authhandlers.go

```diff
@@ -318,7 +318,7 @@ func (h *AuthHandlers) UserKeyAuth(conn ssh.ConnMetadata, key ssh.PublicKey) (*s
 
 	// check if the user has permission to log into the node.
 	switch {
-	case h.c.Component == teleport.ComponentForwardingNode:
+	case h.c.Component == teleport.ComponentForwardingNode, conn.User() == teleport.SSHSessionJoinPrincipal:
```

Comment:
What was the issue with checking RBAC here?

Reply:
The point of this PR is to skip the regular RBAC checks for node access which are also the ones used to decide if you can start a session. The reasoning behind this is that you should be able to join a session (perhaps as a witness) if you have an applicable join policy even though you don't have access to start sessions on the node.